server/internal/store/usage.go

```diff
@@ -45,3 +45,37 @@ func (s *Store) FindUsages() ([]*Usage, error) {
 	}
 	return usages, nil
 }
+
+// UsageByGroup represents the aggregated usage data grouped by API key, user, and model.
+type UsageByGroup struct {
+	APIKeyID string
+	UserID   string
+	ModelID  string
+
+	TotalRequests           int64
+	TotalPromptTokens       int64
+	TotalCompletionTokens   int64
+	AverageLatency          float64
+	AverageTimeToFirstToken float64
+}
+
+// GetUsagesByGroups returns the aggregated usage data by groyps for the given tenant and time range.
+func (s *Store) GetUsagesByGroups(tenantID string, start, end int64) ([]*UsageByGroup, error) {
+	var us []*UsageByGroup
+	if err := s.db.Model(&Usage{}).
+		Select(
+			`api_key_id`, `user_id`, `model_id`,
```

Comment:
Btw, ideally we would like to convert API key IDs to API key name and (internal) user ID to external user IDs. No need to do that now, though.